UnitTests/CommonTestUtils/ConfigureJoinableTaskFactoryAttribute.cs

```diff
@@ -43,8 +52,129 @@ public void BeforeTest(ITest test)
 
         public void AfterTest(ITest test)
         {
-            ThreadHelper.JoinableTaskContext?.Factory.Run(() => ThreadHelper.JoinPendingOperationsAsync());
-            ThreadHelper.JoinableTaskContext = null;
+            try
+            {
+                ThreadHelper.JoinableTaskContext?.Factory.Run(() => ThreadHelper.JoinPendingOperationsAsync());
+                ThreadHelper.JoinableTaskContext = null;
+                if (_denyExecutionSynchronizationContext != null)
+                {
+                    SynchronizationContext.SetSynchronizationContext(_denyExecutionSynchronizationContext.UnderlyingContext);
+                    _denyExecutionSynchronizationContext.ThrowIfSwitchOccurred();
+                    _denyExecutionSynchronizationContext.Dispose();
```

Comment:
`_denyExecutionSynchronizationContext` should always be disposed. It has to be done in an outer try-finally block.